File: app/src/main/java/de/metzgore/rbtvschedule/api/RBTVScheduleApi.java
package de.metzgore.rbtvschedule.api;

import android.arch.lifecycle.LiveData;

import de.metzgore.rbtvschedule.data.DailySchedule;
import de.metzgore.rbtvschedule.data.WeeklySchedule;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface RBTVScheduleApi {
    @GET("/api/1.0/schedule/schedule.json")
    LiveData<ApiResponse<WeeklySchedule>> scheduleOfCurrentWeek();

    @GET("/api/1.0/schedule/{year}/{month}/{day}.json")
    LiveData<ApiResponse<DailySchedule>> scheduleOfDay(@Path("year") int year, @Path("month") String month, @Path("day") String day);
}

File: app/src/main/java/de/metzgore/beansplan/api/RbtvScheduleApi.kt
package de.metzgore.beansplan.api

import androidx.lifecycle.LiveData
import de.metzgore.beansplan.data.WeeklyScheduleResponse
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface RbtvScheduleApi {
    @GET("/v1/schedule/normalized")
    fun getScheduleFromTimestamp(@Query("startDay") startDay: Int): LiveData<ApiResponse<WeeklyScheduleResponse>>
}
